contracts/instruments/StakedPut.sol

```diff
@@ -0,0 +1,326 @@
+    uint8 private constant venueID = 1;
+
+    ProtocolAdapterTypes.OptionType public constant optionType = ProtocolAdapterTypes.OptionType.Put;
+    address public immutable ethAddress;
+    address public immutable wbtcAddress;
+    address public immutable underlying;
+    address public immutable strikeAsset;
+    address public immutable collateralAsset;
+    address public immutable optionsAddress;
+
+
+    constructor(address _factory, address payable _uniswapAdapterAddress, address _ethWbtcPair, address _ethAddress, address _wbtcAddress, address _wbtcOptionsAddress, address _collateralAsset, address _priceFeed) {
+        require(_factory != address(0), "!_factory");
+        require(_uniswapAdapterAddress != address(0), "!_uniswapAdapter");
+        require(_ethWbtcPair != address(0), "!_ethWbtcPair");
+        require(_ethAddress != address(0), "!_eth");
+        require(_wbtcAddress != address(0), "!_wbtc");
+        require(_wbtcOptionsAddress != address(0), "!_wbtcOptions");
+        require(_collateralAsset != address(0), "!_collateral");
+        require(_priceFeed != address(0), "!_priceFeed");
+
+        ethAddress = _ethAddress;
+        wbtcAddress = _wbtcAddress;
+        underlying = _wbtcAddress;
+        strikeAsset = _wbtcAddress;
+        collateralAsset = _collateralAsset;
+        priceProvider = AggregatorV3Interface(_priceFeed);
+        IRibbonFactory factoryInstance = IRibbonFactory(_factory);
+        iUniswapAdapter = IAmmAdapter(_uniswapAdapterAddress);
+        uniswapAdapterAddress = _uniswapAdapterAddress;
+        ethWbtcPair = IUniswapV2Pair(_ethWbtcPair);
+        address adapterAddress = factoryInstance.getAdapter(adapterName);
+        require(adapterAddress != address(0), "Adapter not set");
+        options = IHegicOptions(_wbtcOptionsAddress);
+        factory = factoryInstance;
+        adapter = IProtocolAdapter(adapterAddress);
+        optionsAddress = _wbtcOptionsAddress;
+
+    }
+
+    function initialize() external initializer {
+    }
+
+    function getName() public pure returns(string memory)
+    {
+        return instrumentName;
+    }
+
+    function getCurrentPrice() public view returns(uint256)
+    {
+        (, int256 latestPrice, , , ) = priceProvider.latestRoundData();
+        uint256 currentPrice = uint256(latestPrice);
+        return currentPrice.mul(10000000000);
+    }
+
+    function getInputs(address inputToken, uint256 amt, string memory exchangeName) public view returns(uint256 wbtcSize, uint256 expDigg, uint256 tradeAmt, uint256 premium, uint256 totalCost,uint256 currentPrice, uint256 expiry)
+    {
+        require(inputToken == ethAddress, 'invalid input token');
+        wbtcSize = iUniswapAdapter.expectedWbtcOut(amt, exchangeName);
+
+        (expDigg, tradeAmt) = iUniswapAdapter.expectedDiggOut(wbtcSize, exchangeName);
+
+        //set expiry to a month from now
+        //set strike to atm
+        expiry = block.timestamp + month;

```

Comment:
Can we double check that the Hegic option purchase won't revert because of using an expiry a month from now? This is because Hegic has the limitation of having options <4 weeks.

We have to test the case `block.timestamp + month - 1 second` in our tests